model_analyzer/config/generate/config_parameters.py

```diff
@@ -53,12 +54,146 @@ def get_type(self, name: str) -> ParameterType:
     def get_category(self, name: str) -> ParameterCategory:
         return self._parameters[name].category
 
-    def get_range(self, name: str) -> Tuple[int, int]:
+    def get_range(self, name: str) -> Tuple[Optional[int], Optional[int]]:
         return (self._parameters[name].min_range, self._parameters[name].max_range)
 
-    def get_list(self, name: str) -> List[Any]:
+    def get_list(self, name: str) -> Optional[List[Any]]:
         return self._parameters[name].enumerated_list
 
+    def populate_search_parameters(
+        self,
+        config: ConfigCommandProfile,
+        parameters: Dict[str, Any],
+        model_config_parameters: Dict[str, Any],
+    ) -> None:
+        self._populate_parameters(config, parameters)
+        self._populate_model_config_parameters(config, model_config_parameters)
+
+    def _populate_parameters(
+        self,
+        config: ConfigCommandProfile,
+        parameters: Dict[str, Any],
+    ) -> None:
+        self._populate_parameter(
+            parameter_name="batch_sizes",
+            rcs_min_value=config.run_config_search_min_model_batch_size,
+            rcs_max_value=config.run_config_search_max_model_batch_size,
+            parameter_list=parameters["batch_sizes"],
+        )
+        # TODO: Figure out how to use request rate
+        self._populate_parameter(
+            parameter_name="concurrency",
+            rcs_min_value=config.run_config_search_min_concurrency,
+            rcs_max_value=config.run_config_search_max_concurrency,
+            parameter_list=parameters["concurrency"],
+        )
+
+    def _populate_parameter(
+        self,
+        parameter_name: str,
+        rcs_min_value: Optional[int] = None,
+        rcs_max_value: Optional[int] = None,
+        parameter_list: Optional[List[int]] = None,
+    ) -> None:
+        ptype = self._determine_parameter_type(parameter_name)
+
+        if parameter_list:
+            self._add_parameter(
+                name=parameter_name,
+                ptype=ptype,
+                category=ParameterCategory.LIST,
+                enumerated_list=parameter_list,
+            )
+        else:
+            category = self._determine_parameter_category(parameter_name)
+
+            if category == ParameterCategory.EXPONENTIAL:
```

Comment:
This logic appears to be very trusting that things are done right elsewhere. Do we want to add safeguards?

If a parameter list is passed, do the RCS values get silently ignored? I thought all four parameters (including parameter_list) are passed elsewhere, which means the RCS values would be ignored. If a parameter list is not passed, it feels like the error is a bit misleading, since what if it's not passed but the parameter isn't supposed to be exponential/linear. I think some refactoring needs to be done here. I think this can exist as one function, but with increased documentation to understand what is happening and safeguarding to prevent logic errors.